src/code-components/RichTextArea/Editor.tsx

```diff
@@ -13,6 +18,9 @@ export interface EditorProps {
   onFocus?: (range: Range, source: EmitterSource) => void;
   onKeyDown?: (event: KeyboardEvent) => void;
   onKeyUp?: (event: KeyboardEvent) => void;
+  onError?: (error: { message?: string } | undefined) => void;
+  onLoading?: (loading: boolean) => void;
+  onDrop?: (type: string, name: string, blob: Blob) => Promise<string>;
```

Comment:
When is this called? Whenever anything is dropped? Or only when image is dropped?

What if I copy paste an image (not drop it). Will this also be called?

Why don't we have one function for both goals?


```suggestion
  onImageUpload?: (type: string, name: string, blob: Blob) => Promise<string>;
```

Reply:
True. It will also be called on paste. I'll change the name